src/__tests__/field/captcha/third_party_captcha.test.jsx

```diff
@@ -0,0 +1,231 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import I from 'immutable';
+import * as l from '../../../core/index';
+import { ThirdPartyCaptcha } from '../../../field/captcha/third_party_captcha';
+
+const createLockMock = ({
+  provider = 'auth0',
+  required = true,
+  siteKey = '',
+  clientSubdomain = ''
+} = {}) =>
+  I.fromJS({
+    id: '__lock-id__',
+    core: {
+      captcha: { provider, siteKey, clientSubdomain, required: required }
+    }
+  });
+
+describe('ThirdPartyCaptcha', () => {
+  describe('recaptchav2', () => {
+    let shallowWrapper;
+    beforeAll(() => {
+      const lockMock = createLockMock({
+        provider: 'recaptcha_v2',
+        siteKey: 'mySiteKey'
+      });
+
+      const captcha = l.captcha(lockMock);
+      shallowWrapper = shallow(
+        <ThirdPartyCaptcha
+          provider={captcha.get('provider')}
+          sitekey={captcha.get('siteKey')}
+          clientSubdomain={captcha.get('clientSubdomain')}
+          hl={'en'}
+          isValid={true}
+          value={undefined}
+        />
+      ).instance();
+      shallowWrapper.componentDidMount();
+    });
+
+    it('should have correct renderParams', () => {
+      const { renderParams } = shallowWrapper;
+      expect(renderParams).toMatchObject({
+        sitekey: 'mySiteKey',
+        callback: expect.any(Function),
+        'expired-callback': expect.any(Function),
+        'error-callback': expect.any(Function)
+      });
+      expect(renderParams.language).toBeUndefined();
+      expect(renderParams.theme).toBeUndefined();
+      expect(Object.keys(renderParams)).toHaveLength(4);
+    });
+  });
+
+  describe('friendly captcha', () => {
+    let shallowWrapper;
+    beforeAll(() => {
+      const lockMock = createLockMock({
+        provider: 'friendly_captcha',
+        siteKey: 'mySiteKey'
+      });
+
+      const captcha = l.captcha(lockMock);
+      shallowWrapper = shallow(
+        <ThirdPartyCaptcha
+          provider={captcha.get('provider')}
+          sitekey={captcha.get('siteKey')}
+          clientSubdomain={captcha.get('clientSubdomain')}
+          hl={'en'}
+          isValid={true}
+          value={undefined}
+        />
+      ).instance();
+      shallowWrapper.componentDidMount();
+    });
+
+    it('should have correct renderParams', () => {
+      const { renderParams } = shallowWrapper;
+      expect(renderParams).toMatchObject({
+        sitekey: 'mySiteKey',
+        doneCallback: expect.any(Function),
+        errorCallback: expect.any(Function),
+        language: 'en'
+      });
+      expect(renderParams.theme).toBeUndefined();
+      expect(renderParams.callback).toBeUndefined();
+      expect(renderParams['expired-callback']).toBeUndefined();
+      expect(renderParams['error-callback']).toBeUndefined();
+      expect(Object.keys(renderParams)).toHaveLength(4);
+    });
+  });
```

Comment:
Based on my other comment, I believe we should rework this test (as well as comparable tests), as it makes more sense to test what gets passed to the provider rather then test the implementation detail that we hold renderParams as a property on the component.

Additionally, we want to use `.toEqual` as that should ensure we do not need all the rest.

```
expect(renderParams).toEqual({
        sitekey: 'mySiteKey',
        doneCallback: expect.any(Function),
        errorCallback: expect.any(Function),
        language: 'en'
});
```

Note that we want to use this on the argument passed to the provider, so we will need something like:

```
expect(provider.render.mock.calls[0][1]).toEqual({
        sitekey: 'mySiteKey',
        doneCallback: expect.any(Function),
        errorCallback: expect.any(Function),
        language: 'en'
})
```

Note the above is just to show what we could do, and not a copy pasteable solution.

Reply:
Yeah, there were definitely a few mocking hoops to jump through before I could do something like that but it should be what you're looking for now.